relay-sampling/src/lib.rs

```diff
@@ -2182,408 +2572,288 @@ mod tests {
+                id: RuleId(1),
+                time_range: Default::default(),
+                decaying_fn: Default::default(),
+            },
+            SamplingRule {
+                condition: and(vec![glob("trace.environment", &["*dev*"])]),
+                sampling_value: SamplingValue::SampleRate { value: 1.0 },
+                ty: RuleType::Trace,
+                id: RuleId(2),
+                time_range: Default::default(),
+                decaying_fn: Default::default(),
+            },
+            SamplingRule {
+                condition: and(vec![eq("event.transaction", &["foo"], true)]),
+                sampling_value: SamplingValue::Factor { value: 2.0 },
+                ty: RuleType::Transaction,
+                id: RuleId(3),
+                time_range: Default::default(),
+                decaying_fn: Default::default(),
+            },
+            SamplingRule {
+                condition: and(vec![
+                    glob("trace.release", &["1.1.1"]),
+                    eq("trace.user.segment", &["vip"], true),
+                ]),
+                sampling_value: SamplingValue::SampleRate { value: 0.5 },
+                ty: RuleType::Trace,
+                id: RuleId(4),
+                time_range: Default::default(),
+                decaying_fn: Default::default(),
+            },
+            SamplingRule {
+                condition: and(vec![
+                    eq("trace.release", &["1.1.1"], true),
+                    eq("trace.environment", &["prod"], true),
+                ]),
+                sampling_value: SamplingValue::Factor { value: 1.5 },
+                ty: RuleType::Trace,
+                id: RuleId(5),
+                time_range: Default::default(),
+                decaying_fn: Default::default(),
+            },
+            SamplingRule {
+                condition: and(vec![]),
+                sampling_value: SamplingValue::SampleRate { value: 0.02 },
+                ty: RuleType::Trace,
+                id: RuleId(6),
+                time_range: Default::default(),
+                decaying_fn: Default::default(),
+            },
+        ]);
 
-        let old_release = Event {
-            ty: Annotated::new(EventType::Transaction),
-            release: Annotated::new("1.1.0".to_string().into()),
-            environment: Annotated::new("testing".to_string()),
-            ..Default::default()
-        };
-        let result = rules.get_matching_event_rule(&old_release, None, Utc::now());
-        assert_eq!(result.unwrap().id, RuleId(5), "Did not match expected rule");
+        // early return of first rule
+        let event = mocked_event(EventType::Transaction, "healthcheck", "1.1.1", "testing");
+        let dsc =
+            mocked_dynamic_sampling_context("root_transaction", "1.1.1", "debug", "vip", "user-id");
+        let result = match_against_rules(&config, &event, &dsc, Utc::now());
+        transaction_match!(result, 0.1, event, 1);
```

Comment:
nit: Putting `assert` in the name might help clarify the intent of the test (same goes for other macros).
```suggestion
        assert_transaction_match!(result, 0.1, event, 1);
```